ui/dashboards/src/components/GridLayout/GridItemContent.tsx

```diff
@@ -24,5 +24,17 @@ export interface GridItemContentProps {
 export function GridItemContent(props: GridItemContentProps) {
   const { panelGroupItemId } = props;
   const panelDefinition = usePanel(panelGroupItemId);
-  return <Panel definition={panelDefinition} panelGroupItemId={panelGroupItemId} />;
+  const { isEditMode } = useEditMode();
+  const { openEditPanel, openDeletePanelDialog } = usePanelActions(panelGroupItemId);
+
+  // Provide actions to the panel when in edit mode
+  let editHandlers: PanelProps['editHandlers'] = undefined;
+  if (isEditMode) {
+    editHandlers = {
+      onEditPanelClick: openEditPanel,
+      onDeletePanelClick: openDeletePanelDialog,
+    };
+  }
+
+  return <Panel definition={panelDefinition} editHandlers={editHandlers} />;
```

Comment:
I wonder if we want to call these `actionHandlers` for a future when we might have other actions to handle (duplicate?)

Reply:
But also this seems good for now, we always do that if/when that day comes

Reply:
in the future, it would also be cool if users can customize what icons they want there